python/tests/integration/arcticdb/version_store/test_basic_version_store.py

```diff
@@ -210,6 +210,66 @@ def test_prune_previous_versions_multiple_times(lmdb_version_store, symbol):
     assert len([ver for ver in lmdb_version_store.list_versions() if not ver["deleted"]]) == 1
 
 
+def test_prune_previous_versions_write_batch(lmdb_version_store):
+    """Verify that the batch write method correctly prunes previous versions when the corresponding option is specified.
+    """
+    # Given
+    lib = lmdb_version_store
+    lib_tool = lib.library_tool()
+    sym1 = "test_symbol1"
+    sym2 = "test_symbol2"
+    df0 = pd.DataFrame({"col_0": ["a", "b"]}, index=pd.date_range("2000-01-01", periods=2))
+    df1 = pd.DataFrame({"col_0": ["c", "d"]}, index=pd.date_range("2000-01-03", periods=2))
+
+    # When
+    lib.batch_write([sym1, sym2], [df0, df0])
+    lib.batch_write([sym1, sym2], [df1, df1], prune_previous_version=True)
+
+    # Then - only latest version and keys should survive
+    assert len(lib.list_versions(sym1)) == 1
+    assert len(lib.list_versions(sym2)) == 1
+    assert len(lib_tool.find_keys(KeyType.TABLE_INDEX)) == 2
+    assert len(lib_tool.find_keys(KeyType.TABLE_DATA)) == 2
+    # Then - we got 3 version keys per symbol: version 0, version 0 tombstone, version 1
+    version_keys = lib_tool.find_keys(KeyType.VERSION)
+    keys_for_sym1 = [k for k in version_keys if k.id == sym1]
+    keys_for_sym2 = [k for k in version_keys if k.id == sym2]
+    assert len(keys_for_sym1) == 3
+    assert len(keys_for_sym2) == 3
+    # Then - we got 4 symbol keys: 2 for each of the writes
+    assert len(lib_tool.find_keys(KeyType.SYMBOL_LIST)) == 4
+
+
+def test_prune_previous_versions_append_batch(lmdb_version_store):
```

Comment:
What about a similar test for batch write metadata?

Reply:
sounds plausible

Reply:
I don't think it should delete any index or data keys, but I want the expected behaviour to be codified in a test

Reply:
If prune_previous_versions is enabled for the write_metadata operation and there are existing previous versions, the expected behavior would be to create a new version that includes the previous data along with the new metadata. Additionally, any previous versions that are no longer needed would be removed, along with any unused data or index keys.
In summary, the operation would create a new version with the combined data and metadata, while removing unnecessary previous versions and associated keys.

Is it not what you expect @alexowens90 ?

Reply:
Yes, this is what I expect